food-search-app/src/main/java/com/example/foodsearchapp/controller/AuthController.java

```diff
@@ -0,0 +1,118 @@
+
+import com.example.foodsearchapp.config.token.Utility;
+import com.example.foodsearchapp.model.EStatus;
+import com.example.foodsearchapp.model.Status;
+import com.example.foodsearchapp.model.User;
+import com.example.foodsearchapp.pojo.JsonWebTokenResponse;
+import com.example.foodsearchapp.pojo.LoginRequest;
+import com.example.foodsearchapp.pojo.MessResponse;
+import com.example.foodsearchapp.pojo.RegistrationRequest;
+import com.example.foodsearchapp.repo.StatusRepo;
+import com.example.foodsearchapp.repo.UserRepo;
+import com.example.foodsearchapp.service.UserDetailsImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/auth")
+@CrossOrigin(origins = "*", maxAge = 3600)
+public class AuthController {
+
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    UserRepo userRepo;
+
+    @Autowired
+    StatusRepo statusRepo;
+
+    @Autowired
+    PasswordEncoder passwordEncoder;
+
+    @Autowired
+    Utility utility;
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authUser(@RequestBody LoginRequest loginRequest) {
+
+        Authentication auth = authenticationManager
+                .authenticate(new UsernamePasswordAuthenticationToken(
+                        loginRequest.getUsername(),
+                        loginRequest.getPass()));
+
+        SecurityContextHolder.getContext().setAuthentication(auth);
+        String jsonWebToken = utility.generateJsonWebToken(auth);
+
+        UserDetailsImpl userDetails = (UserDetailsImpl) auth.getPrincipal();
+        List<String> statuses = userDetails.getAuthorities().stream()
+                .map(GrantedAuthority::getAuthority)
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(new JsonWebTokenResponse(jsonWebToken,
+                userDetails.getId(),
+                userDetails.getUsername(),
+                userDetails.getEmail(),
+                statuses));
+    }
+
+    @PostMapping("/registry")
+    public ResponseEntity<?> registerUser(@RequestBody RegistrationRequest registrationRequest) {
+
+        if (userRepo.existsByUsername(registrationRequest.getUsername())) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(new MessResponse("Error: Username exist"));
```

Comment:
how you spent saved time, which you get, when name Mess instead of message. And also check mess word in vocabulary.